### PR TITLE
Cut out Hive JDBC and Hive Service deps in beeline module

### PR DESCRIPTION
**Pull request by pan3793**
# :mag: Description
## Issue References 🔗

This is the next step of https://github.com/apache/kyuubi/issues/6146, cutting out most Hive deps(excepting `hive-common`) and recovering the skipped tests via minor code tunning.

## Describe Your Solution 🔧

- Drop `hive-jdbc`, `hive-service`, `hive-service-rpc` deps in the beeline module.
- Migrate from `commons-lang` to `commons-lang3` in the beeline module.
- Recover the skipped test `TestClientCommandHookFactory#connectHook`

## Types of changes :bookmark:

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Test Plan 🧪

Pass GA, and manually test to ensure the following error has gone.

Before
```
root@hadoop-master1:/opt/kyuubi# bin/beeline --version
Warn: Not find kyuubi environment file /etc/kyuubi/conf/kyuubi-env.sh, using default ones...
java.lang.NoClassDefFoundError: org/apache/curator/RetryPolicy
```

After
```
root@hadoop-master1:/opt/kyuubi# bin/beeline --version
Connecting to jdbc:hive2://hadoop-master1.orb.local:10000/default;password=hive;user=hive
Connected to: Apache Hive (version 2.3.9)
Driver: Kyuubi Project Hive JDBC Client (version 1.9.0-SNAPSHOT)
Beeline version 1.9.0-SNAPSHOT by Apache Kyuubi
0: jdbc:hive2://hadoop-master1.orb.local:1000>
```

---

# Checklist 📝

- [x] This patch was not authored or co-authored using [Generative Tooling](https://www.apache.org/legal/generative-tooling.html)

**Be nice. Be informative.**

### CODE REVIEW

File: kyuubi-hive-beeline/src/test/java/org/apache/hive/beeline/ProxyAuthTest.java [hunk 1]
```diff
@@ -40,7 +41,7 @@
  * <client-principal>
  */
 public class ProxyAuthTest {
-  private static final String driverName = "org.apache.hive.jdbc.HiveDriver";
+  private static final String driverName = "org.apache.kyuubi.jdbc.KyuubiHiveDriver";
```
Comment (ulysses-you) [inline kyuubi-hive-beeline/src/test/java/org/apache/hive/beeline/ProxyAuthTest.java:44]: we still support `org.apache.hive.jdbc.HiveDriver` right ?

**Comment by ulysses-you** [comment]
lgtm if tests pass

File: kyuubi-hive-beeline/src/test/java/org/apache/hive/beeline/ProxyAuthTest.java (hunk 1 above)
Comment (pan3793) [reply kyuubi-hive-beeline/src/test/java/org/apache/hive/beeline/ProxyAuthTest.java:44]: I think we have no such guarantee, the internally used JDBC driver is not user-faced stuff, we only need to ensure that user could smoothly migrate from Hive Beeline to Kyuubi Beeline without changing the JDBC URL.

**Comment by pan3793** [comment]
Thanks, merged to master

**Pull request closed without merging**